Header

notlfish opened the pull request:
### What?
Pull request to add the header of the project, following the [project's model](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#how-to-build-the-space-travelers-hub)

### How?
A React Router will be added to the project, and a `Header` component with a logo, title, and a `Navbar`. To make the router work, stubs for the `Rockets`, `Missions`, and `Profile` components will also be added.

### Before merging
The following tasks are to be completed before merging this pull request:
- [x] Add a `Navbar` component that uses NavLinks.
- [x] Add a `Header` component.
- [x] Add stubs for the `Missions`, `Rockets`, and `Profile` components.
- [x] Modify the `App` component to always include the `Header`, and mount either `Missions`, `Rockets`, or `Profile`, depending on the router.
- [x] Style the Header.

notlfish committed:
Add dev-start script

dev-start runs the server process with an env flag to treat linter
errors as warnings instead of treating them as actual errors.

notlfish committed:
Add Header, Navbar and rounting

Add stub Profile, Missions and Rockets components.

Hamzaoutdoors committed:
Add style for the Navbar and links

Hamzaoutdoors reviewed (approved):
## PR APPROVED ✔
@notlfish  Congratulations ✨! Good job working on these assignments we can move on to the next task!

Happy Coding!

notlfish committed:
Fix hamburguer button on medium and small screens

Change styling:
- Aligned navbar brand to be centered.
- Centered header text.
- Aligned nav links to the right.

Hamzaoutdoors committed:
Remove padding for navbar
